controllers/error-controllers.js

```diff
@@ -2,8 +2,8 @@ exports.handle404Errors = (request, response, next) => {
   response.status(404).send({ message: 'Not Found'})
 }
 
-exports.handlePsqlErrors = (err, request, response, next) => {
-	(err.code === '23502' || err.code === '22P02') ?
+exports.handlePsqlErrors = (err, request, response, next) => { 
+	(err.code === '23502' || err.code === '22P02' || err.code === '23503') ?
 		response.status(400).send({ message : 'Bad Request' }) :
```

Comment:
an error code of 23503 would actually be a 404 error - so we should handle it a bit differently and send back a different response - this is because the code denotes that a foreign key was unable to match to any data in the foreign table , similar to a "not found"